### PR TITLE
Update balance

### LINKED ISSUE #1209: [Task]: Remove expired outputs from potentially_locked_outputs
### Task description

Outputs where the wallet has the address in the address unlock condition and which are expired, still show up in the potentially_locked_outputs in the balance. They shouldn't show up anymore, because we can't claim them(or own it anyways if the expiraton address is ours).

### Requirements

-

### Acceptance criteria

-

### Creation checklist

- [ ] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Add amounts from Alias and Foundry outputs, don't add unlockable outputs in potentially_locked_outputs
Also update AccountBalance struct

## Links to any relevant issues

Fixes #1209, #1246

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

Cli wallet with an expired output

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by thibault-martinez — DISMISSED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -21,6 +21,8 @@ iota-client = { git = "https://github.com/iotaledger/iota.rs", rev = "09e8d0f296
 iota-crypto = { version = "0.12.1", default-features = false, features = [ "std", "chacha", "blake2b", "ed25519", "random", "slip10", "bip39", "bip39-en" ] }
 log = { version = "0.4.17", default-features = false }
 packable = { version = "0.4.0", default-features = false, features = [ "serde", "primitive-types" ] }
+# remove again when U256Dto is imported from bee https://github.com/iotaledger/bee/pull/1454
+prefix-hex = { version = "0.4.0", default-features = false, features = [ "primitive-types" ] }
```
Comment (thibault-martinez) [inline Cargo.toml]: It's been merged

File: src/message_interface/dtos.rs [hunk 2]
```diff
@@ -113,20 +114,59 @@ impl From<&AddressWithUnspentOutputs> for AddressWithUnspentOutputsDto {
     }
 }
 
-/// Dto for the balance of an account, returned from [`crate::account::handle::AccountHandle::sync()`] and
-/// [`crate::account::handle::AccountHandle::balance()`].
-#[derive(Clone, Debug, Default, Eq, PartialEq, Serialize, Deserialize)]
-pub struct AccountBalanceDto {
+/// Base coin fields for [`AccountBalance`]
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct BaseCoinDto {
```
Comment (thibault-martinez) [inline src/message_interface/dtos.rs]: I think I'd call it `BaseCoinBalanceDto` for consistency with `NativeTokensBalanceDto`

File: src/message_interface/dtos.rs [hunk 3]
```diff
@@ -113,20 +114,59 @@ impl From<&AddressWithUnspentOutputs> for AddressWithUnspentOutputsDto {
...
+
+/// Base coin fields for [`AccountBalanceDto`]
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct NativeTokensBalanceDto {
+    /// Token id
+    #[serde(rename = "tokenId")]
+    pub token_id: TokenIdDto,
+    /// Total amount
+    pub total: U256Dto,
+    /// Balance that can currently be spend
```
Comment (thibault-martinez) [inline src/message_interface/dtos.rs]:
```suggestion
    /// Balance that can currently be spent
```

File: src/account/types/mod.rs [hunk 4]
```diff
@@ -48,21 +48,37 @@ pub struct AccountBalance {
     pub potentially_locked_outputs: HashMap<OutputId, bool>,
 }
 
-impl Default for AccountBalance {
+/// Base coin fields for [`AccountBalance`]
+#[derive(Clone, Debug, Default, Eq, PartialEq, Serialize, Deserialize)]
+pub struct BaseCoin {
```
Comment (thibault-martinez) [inline src/account/types/mod.rs]: Also rename this ?

**Review by samuel-rufi — DISMISSED** [review]

File: src/account/operations/balance.rs [hunk 5]
```diff
@@ -189,3 +236,28 @@ impl AccountHandle {
...
+
+    for balance in balances {
+        total_balance.base_coin.total += balance.base_coin.total;
+        total_balance.base_coin.available += balance.base_coin.available;
+        total_balance.required_storage_deposit += balance.required_storage_deposit;
+        total_balance.nfts.extend(balance.nfts.into_iter());
+        total_balance.aliases.extend(balance.aliases.into_iter());
+        total_balance.foundries.extend(balance.foundries.into_iter());
+        for native_token_balance in &balance.native_tokens {
+            if let Some(total_native_token_balance) = total_balance
```
Comment (thibault-martinez) [inline src/account/operations/balance.rs:251]: You never really add the native tokens to `total_balance` ?

Comment (Thoralf-M) [reply src/account/operations/balance.rs:251]: Fixed

**Review by thibault-martinez — DISMISSED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**